xds/src/test/java/io/grpc/xds/XdsServerTestHelper.java

```diff
@@ -174,12 +179,18 @@ public List<String> getTargets() {
     }
   }
 
+  // Implementation details:
+  // 1. Use `synchronized` in methods where XdsClientImpl uses its own `syncContext`.
+  // 2. Use `serverExecutor` via `execute()` in methods where XdsClientImpl uses watcher's executor.
   static final class FakeXdsClient extends XdsClient {
-    boolean shutdown;
-    SettableFuture<String> ldsResource = SettableFuture.create();
-    ResourceWatcher<LdsUpdate> ldsWatcher;
-    CountDownLatch rdsCount = new CountDownLatch(1);
+    public static final Duration WAIT_TIMEOUT = Duration.ofSeconds(5);
+
+    private boolean shutdown;
+    @Nullable SettableFuture<String> ldsResource = SettableFuture.create();
+    @Nullable ResourceWatcher<LdsUpdate> ldsWatcher;
+    private volatile CountDownLatch rdsCount = new CountDownLatch(1);
```

Comment:
Don't make this `volatile`. Changing the expected count can't be done concurrently with other threads.

Reply:
I'm saying, "Any change to the `rdsCount` reference can't be done concurrently with any other thread." It doesn't matter who's doing what. There must already be a happens-before in place when changing the `rdsCount` reference, thus `volatile` just hides any races from TSAN.